### PR TITLE
Improve docs

### LINKED ISSUE #75: Document how arguments are passed to the functions
Please document how parameters are passed to the functions.

Also please document how strict is type checking and how much of the type checking is already implemented.

### LINKED ISSUE #78: Document response handling
Create a section in the documentation for response handling.
It should include:

- Serialization (From the current quickstart section)
- How connexion handles response objects
- How to return status code
- After #77 is merged, how to return headers

### LINKED ISSUE #167: Include a link to the RTD in the README.
Include a link to the documentation in read the docs in the README and make sure all the information the README is also on the documentation

### PR DESCRIPTION
**Pull request by jmcs**
Fixes #167 
Fixes #78 
Fixes #75.

# Description

Changes proposed in this pull request:

 - Large documentation update

### CODE REVIEW

**Comment by jmcs** [comment]
@rafaelcaricio @hjacobs @valgog please check if I didn't forget anything

File: docs/response.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
+Response Handling
+=================
+
+Response Serialization
+----------------------
+If the endpoint returns a `Response` object this response will be used as is.
+
+Otherwise, and by default and if the specification defines that a endpoint
```
Comment (rafaelcaricio) [inline docs/response.rst]: small typo "a**n** endpoint"

File: docs/response.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+produces only json, connexion will automatically serialize the return value
```
Comment (rafaelcaricio) [inline docs/response.rst]: Maybe we should use always Connexion in uppercase?!

File: docs/response.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+for you and set the right content type in the HTTP header.
+
+If the endpoint produces a single non json mimetype then connexion will
```
Comment (rafaelcaricio) [inline docs/response.rst]: JSON

File: docs/response.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+automatically set the right content type in the HTTP header.
+
+Returning a status codes
```
Comment (rafaelcaricio) [inline docs/response.rst]: Typo, remove the "a"

File: docs/response.rst [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+The other is returning it as second return value in the response. For example
+
+.. code-block:: python
+
+    def my_endpoint():
+        return 'Not Found', 404
+
+Returning Headers
+-----------------
+There are two ways to return headers from you endpoints.
```
Comment (rafaelcaricio) [inline docs/response.rst]: typo "your"

**Comment by rafaelcaricio** [comment]
:+1: LGTM.

**Comment by LappleApple** [comment]
Go ahead and merge, then I'll edit. I'm not a great fan of the editing interface for GitHub; much easier for me to reread the README and find things that way. :)

**Comment by hjacobs** [comment]
Yeah, documentation :book: :smile:

**Pull request merged**